sdk/state/payment.go

```diff
@@ -95,9 +104,13 @@ func (c *Channel) validatePayment(ca CloseAgreement) (err error) {
 	if ca.Details.ObservationPeriodTime != c.latestAuthorizedCloseAgreement.Details.ObservationPeriodTime || ca.Details.ObservationPeriodLedgerGap != c.latestAuthorizedCloseAgreement.Details.ObservationPeriodLedgerGap {
 		return fmt.Errorf("invalid payment observation period: different than channel state")
 	}
-	if !c.latestUnauthorizedCloseAgreement.isEmpty() && c.latestUnauthorizedCloseAgreement.Details != ca.Details {
+	if !c.latestUnauthorizedCloseAgreement.isEmpty() && !ca.Details.Equal(c.latestUnauthorizedCloseAgreement.Details) {
 		return fmt.Errorf("close agreement does not match the close agreement already in progress")
 	}
+	if ca.Details.ConfirmingSigner.Address() != c.localSigner.Address() &&
```

Comment:
❓ oh wait, looks like we can compare them directly. What's the reason we're comparing each field directly in `validateClose` in `close.go` and not here?

Reply:
In the close case we want to validate the fields with different values, like the observation periods are fixed rather than equal to the last detail.